ProjectMars/StepDefinitions/SkillFeatureStepDefinitions.cs

```diff
@@ -0,0 +1,160 @@
+using NUnit.Framework;
+using OpenQA.Selenium;
+using ProjectMars.Pages;
+using ProjectMars.Utilities;
+using System;
+using TechTalk.SpecFlow;
+
+namespace ProjectMars.StepDefinitions
+{
+
+    
+   [Binding]
+    public class SkillFeatureStepDefinitions:CommonDriver
+    {
+        LoginPage loginPageObj;
+        ProfilePage profilePageObj;
+
+        [Given(@"I logged into Mars website sucessfully")]
+        public void GivenILoggedIntoMarsWebsiteSucessfully()
+        {
+
+            loginPageObj = new LoginPage(driver);
+            loginPageObj.LoginActions(driver, "antususansunila@gmail.com", "Ammuj@me5");
+        }
+
+        [Given(@"I navigate to Skill page")]
+        public void GivenINavigateToSkillPage()
+        {
+            profilePageObj = new ProfilePage(driver);
+            profilePageObj.GotoSkillsTab(driver);
+        }
+
+        [When(@"I add a new '([^']*)' and '([^']*)'")]
+        public void WhenIAddANewAnd(string p0, string p1)
+        {
+            profilePageObj.ResetSkillRow(driver);
+            profilePageObj.CreateSkillsRecords(driver, p0, p1);
+        }
+
+        [Then(@"table should have the new '([^']*)' and  '([^']*)'")]
+        public void ThenTableShouldHaveTheNewAnd(string p0, string p1)
+        {
+            string newSkill = profilePageObj.GetSkill(driver);
+            string newSkillLevel = profilePageObj.GetSkillLevel(driver);
+           Assert.That(newSkill == p0, "Skill has nott been created sucessfully");
+             Assert.That(newSkillLevel == p1, "Level has not been created sucesssfully");
+        }
+
+
+        [When(@"I add an invalid skill'([^']*)' and valid '([^']*)'")]
+        public void WhenIAddAnInvalidSkillAndValid(string p0, string p1)
+        {
+            profilePageObj.ResetSkillRow(driver);
+            profilePageObj.CreateSkillsRecords(driver, p0, p1);
+        }
+
+        [Then(@"the record has not be added as '([^']*)' sucessfully")]
+        public void ThenTheRecordHasNotBeAddedAsSucessfully(string p2)
+        {
+            string actualResult1 = driver.FindElement(By.XPath("//*[@class='ns-box-inner']")).Text;
```

Comment:
please move this within the page, create a method that returns the value and then use it! best practice to contain page logic within pages